src/library/scala/collection/IterableOnce.scala

```diff
@@ -1035,7 +1037,11 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
   def copyToArray[B >: A](xs: Array[B], start: Int, len: Int): Int = {
```

Comment:
Better update the scala doc for the new behavior.

Reply:
I think we'll need a global documentation of out-of-bounds behaviors, as requested in the linked tickets.

This change satisfies the documented behavior, but we need a phrase like "all methods in this package throw on null input", as we read in Javadoc.